googletest/src/matchers/mod.rs

```diff
@@ -87,3 +87,23 @@ pub use str_matcher::{
 };
 pub use subset_of_matcher::subset_of;
 pub use superset_of_matcher::superset_of;
+
+// Types and functions used by macros matchers.
+// Do not use directly.
+// We may perform incompatible changes without major release. These elements
+// should only be used through their respective macros.
+#[doc(hidden)]
+pub mod __internal_macro_only {
+    pub use super::all_matcher::internal::AllMatcher;
+    pub use super::any_matcher::internal::AnyMatcher;
+    pub use super::conjunction_matcher::ConjunctionMatcher;
```

Comment:
Nit: This struct and the one imported on the next line aren't involved in macros (as suggested above), but rather in the `and` respectively `or` methods of the `Matcher` trait itself.

Reply:
Yes, the name was a little awkward in the matcher.rs. Updated to __internal_unstable_do_not_depend_on_these. Maybe we could go back to __internal_macro_only once impl return in traits is implemented (but maybe also not worth it).

Reply:
Probably `__internal_only` would convey the same meaning without as many words, but this is also fine :-)